lib/mirage.ml

```diff
@@ -724,7 +810,7 @@ let udpv4_socket_conf ipv4_key = object
   method libraries =
     Key.match_ Key.(value target) @@ function
     | `Unix | `MacOSX -> [ "tcpip.udpv4-socket" ]
-    | `Xen | `Virtio | `Ukvm -> failwith "No socket implementation available for unikernel"
+    | `Xen | `Virtio | `Ukvm | `Qubes  -> failwith "No socket implementation available for unikernel"
```

Comment:
It seems the behavior of `Key.match_` and a pattern is subtly different in the evaluation results from `Key.if_ is_unix` -- replacing this pattern match as suggested results in an immediate failure for any `mirage configure` with the `"No socket implementation"` message.  Possibly both branches of the `if` are evaluated in that case (and only the one which matches the actual value used), but this isn't done with `match_`?